tests/unit/test_timeaverage.py

```diff
@@ -0,0 +1,263 @@
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("60") == "2ME"
+    )  # Two months
+
+
+def test__frequency_from_approx_interval_day():
+    assert pymorize.timeaverage._frequency_from_approx_interval("1") == "D"  # One day
+
+
+def test__frequency_from_approx_interval_hour():
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.04167") == "H"
+    )  # Approximately one hour in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.08334") == "2H"
+    )  # Approximately two hours in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.5") == "12H"
+    )  # Half a day in hours
+
+
+def test__frequency_from_approx_interval_minute():
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.000694") == "min"
+    )  # Approximately one minute in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.001388") == "2min"
+    )  # Approximately two minutes in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.020833") == "30min"
+    )  # Approximately half an hour in minutes
+
+
+def test__frequency_from_approx_interval_second():
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.00001157") == "s"
+    )  # Approximately one second in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("0.00002314") == "2s"
+    )  # Approximately two seconds in days
+    assert not (
+        pymorize.timeaverage._frequency_from_approx_interval("0.000694") == "60s"
+    )  # Approximately one minute in seconds, should give back min, since it can round up.
+
+
+def test__frequency_from_approx_interval_millisecond():
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("1.1574e-8") == "ms"
+    )  # Approximately one millisecond in days
+    assert (
+        pymorize.timeaverage._frequency_from_approx_interval("2.3148e-8") == "2ms"
+    )  # Approximately two milliseconds in days
+
+
+def test__invalid_interval():
+    with pytest.raises(ValueError):
+        pymorize.timeaverage._frequency_from_approx_interval("not_a_number")
+
+
+def test__compute_file_timespan_single_chunk():
+    # Create a DataArray with a single chunk
+    time = pd.date_range("2000-01-01", periods=10, freq="D")
+    data = xr.DataArray(np.random.rand(10), dims="time", coords={"time": time})
+    data = data.chunk({"time": 5})  # Single chunk
+
+    assert pymorize.timeaverage._compute_file_timespan(data) == 4
+
+
+def test__compute_file_timespan_multiple_chunks():
+    # Create a DataArray with multiple chunks
+    time = pd.date_range("2000-01-01", periods=30, freq="D")
+    data = xr.DataArray(np.random.rand(30), dims="time", coords={"time": time})
+    data = data.chunk({"time": 10})
+
+    assert pymorize.timeaverage._compute_file_timespan(data) == 9
+
+
+def test__compute_file_timespan_empty_time_dimension():
```

Comment:
since no chunks are defined on data in this test, running this test depends on the outcome of what should happen when running this test: `test__split_by_chunks_no_chunks()`

Reply:
I think raising `ValueError` is fine here, since a timespan only makes sense if the file actually has time.